backend/src/backend/primary/user_session_proxy.py

```diff
@@ -20,12 +20,20 @@ class RadixJobScheduler:
     def __init__(self, name: str, port: int) -> None:
         self._name = name
         self._port = port
-        self._existing_job_names = redis.Redis(host="redis-user-session-state", port=6380, decode_responses=True)
+
+        # redis.Redis does not yet have namespace support - https://github.com/redis/redis-py/issues/12 - need to prefix manually.
+        self._redis_client = redis.Redis(host="redis-user-session", port=6379, decode_responses=True)
+
+    def _get_job_name(self, user_id: str) -> Optional[str]:
+        return self._redis_client.get("user-job-name-" + user_id)
+
+    def _set_job_name(self, user_id: str, job_name: str) -> None:
+        self._redis_client.set("user-job-name-" + user_id, job_name)
```

Comment:
I would suggest extracting the redis access into a small helper class and then utilizing that inside of `RadixJobScheduler`. I feel it is slightly cleaner, *and* it forces all writing/reading to go though a very slim interface.

Something like this:
```
class RedisUserJobs:
    def __init__(self) -> None:
        self._redis_client = redis.Redis.from_url(config.REDIS_URL, decode_responses=True)

    def get_job_name(self, user_id: str) -> Optional[str]:
        return self._redis_client.get("user-job-name:" + user_id)

    def set_job_name(self, user_id: str, job_name: str) -> None:
        self._redis_client.set("user-job-name:" + user_id, job_name)
```

Couple of other points to note:
* Should utilize configuration from `config`
* Don't think there are any hard conventions, but from other code it seems like `:` is often used as a namespace sparator